endurance-page-cache.php

```diff
@@ -152,17 +152,98 @@ function purge_cron( $schedules ) {
 		}
 
 		function option_handler( $option, $old_value, $new_value ) {
-			$exempt_options = array( '_transient', 'cron', 'session', 'sync', 'schedul', 'user_hit_count', 'jetpack_protect', 'rewrite_rules', 'tribe_last', 'wordfence', 'traffic', 'stats' );
-			foreach ( $exempt_options as $exempt_option ) {
-				if ( false !== strpos( $option, $exempt_option ) ) {
-					return;
+			// No need to process if nothing was updated
+			if ( $old_value === $new_value ) {
+				return false;
+			}
+
+			$option_name = str_replace( '-', '_', strtolower( $option ) );
+
+			$exempt_if_equals = array(
+				'active_plugins',
+				'html_type',
+				'fs_accounts',
+				'rewrite_rules',
+				'uninstall_plugins',
+				'wp_user_roles',
+			);
+
+			// If we have an exact match, we can just stop here.
+			if ( in_array( $option, $exempt_if_equals ) ) {
```

Comment:
`in_array()` is easy to read and understand, but is often considerably slower than running an `isset()` -- can we swap out? https://stackoverflow.com/a/13483548

Reply:
@0aveRyan Well, `isset()` checks the array keys, while `in_array()` checks the values. The `in_array()` function is a fraction slower than `strpos()`, but we are also looking for exact matches on a small subset of data. I also think adding a loop through the data would make it slower.

Reply:
@0aveRyan I've replaced the in_array() with array_key_exists() which should have similar performance since it is checking the hash table. I've also added a break in my foreach check later so we can bail early if we find a match.